### PR TITLE
Allow user to view attachments and clear unused attachments

### PR DESCRIPTION
**Pull request by ZeroX-DG**
Hi @ehhc 

I have added a section in setting to allow user to view attachments and clear unused attachments.

Feel free to tell me if you need me to fix something. Thanks :+1:

### CODE REVIEW

File: browser/main/lib/dataApi/attachmentManagement.js [hunk 1]
```diff
@@ -624,6 +624,52 @@ function deleteAttachmentsNotPresentInNote (markdownContent, storageKey, noteKey
   }
 }
 
+function getAttachments (markdownContent, storageKey, noteKey) {
```
Comment (ehhc) [inline browser/main/lib/dataApi/attachmentManagement.js]: The name of the function is missleading.. it only returns certain attachments, won't it?

File: browser/main/lib/dataApi/attachmentManagement.js [hunk 2]
```diff
@@ -728,6 +774,7 @@ module.exports = {
   removeStorageAndNoteReferences,
   deleteAttachmentFolder,
   deleteAttachmentsNotPresentInNote,
+  getAttachments,
```
Comment (ehhc) [inline browser/main/lib/dataApi/attachmentManagement.js]: There is no test for this method..

File: browser/main/modals/PreferencesModal/StoragesTab.js [hunk 3]
```diff
@@ -57,8 +91,61 @@ class StoragesTab extends React.Component {
     e.preventDefault()
   }
 
+  removeAllAttachments (attachments) {
```
Comment (ehhc) [inline browser/main/modals/PreferencesModal/StoragesTab.js]: Maybe this method should also be in the attachmentManagement-file..?

Comment (ehhc) [reply browser/main/modals/PreferencesModal/StoragesTab.js]: And a test for it is required

**Review by ehhc — CHANGES_REQUESTED** [review]

File: browser/main/lib/dataApi/attachmentManagement.js (hunk 1 above)
Comment (ZeroX-DG) [reply browser/main/lib/dataApi/attachmentManagement.js]: Well it get attachment paths for 3 type of attachments:
- Existing attachments
- Existing attachments that are in use
- Existing attachments that aren't in use

What name should I put it? How about `getAttachmentsPath`?

File: browser/main/lib/dataApi/attachmentManagement.js (hunk 2 above)
Comment (ZeroX-DG) [reply browser/main/lib/dataApi/attachmentManagement.js]: I'll add a test for it.

File: browser/main/modals/PreferencesModal/StoragesTab.js (hunk 3 above)
Comment (ZeroX-DG) [reply browser/main/modals/PreferencesModal/StoragesTab.js]: If I'm gonna put it in attachmentManagement, maybe I should change the `getAttachments` function return types to compatible with this. Well I plan to do that anyway, let's do it.

File: browser/main/modals/PreferencesModal/StoragesTab.js [hunk 4]
```diff
@@ -25,6 +27,20 @@ function browseFolder () {
   })
 }
 
+function humanFileSize (bytes) {
```
Comment (ZeroX-DG) [inline browser/main/modals/PreferencesModal/StoragesTab.js]: Note to self: Move this into util file

File: browser/main/lib/dataApi/attachmentManagement.js (hunk 1 above)
Comment (ehhc) [reply browser/main/lib/dataApi/attachmentManagement.js]: hm.. don't know.. if you stay with the name, can you add javadoc explaining what it returns?

**Comment by ZeroX-DG** [comment]
@ehhc I have re-organized the code and added some comments for return types + added test. Tell me if I need to change anything else.

**Comment by ehhc** [comment]
I haven't tested your changes, but at first glance they look good

**Pull request merged**